src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -113,7 +113,7 @@ internal static class ParameterHelpers
             var typeParameters = (object)methodOwner != null ?
                 methodOwner.TypeParameters :
                 default(ImmutableArray<TypeParameterSymbol>);
-            bool allowShadowingNames = binder.Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticLocalFunctions) &&
+            bool allowShadowingNames = binder.Compilation.IsFeatureEnabled(MessageID.IDS_FeatureNameShadowingInNestedFunctions) &&
                 methodOwner?.MethodKind == MethodKind.LocalFunction;
 
```

Comment:
Is this not used for lambdas?

Reply:
Added an assert.

---
In reply to: [251087156](https://github.com/dotnet/roslyn/pull/32754#discussion_r251087156) [](ancestors = 251087156,251082666)